packages/block-library/src/query-title/edit.js

```diff
@@ -34,6 +37,68 @@ export default function QueryTitleEdit( {
 		} ),
 	} );
 
+	const templateTaxonomy = templateSlug?.split( '-' )[ 0 ];
```

Comment:
Todo:

- Is the templateSlug context always defined?
- Update the block so that it is only available in the site editor (in a template)